src/NuGet.Core/NuGet.Protocol/GlobalSuppressions.cs

```diff
@@ -255,15 +255,13 @@
-[assembly: SuppressMessage("Build", "CA2237:Add [Serializable] to ProtocolException as this type implements ISerializable", Justification = "<Pending>", Scope = "type", Target = "~T:NuGet.Protocol.Plugins.ProtocolException")]
 [assembly: SuppressMessage("Build", "CA1001:Type 'RemoteV3FindPackageByIdResource' owns disposable field(s) '_dependencyInfoSemaphore, _httpSource' but is not disposable", Justification = "The inheritance hierarchy of 'RemoteV3FindPackageByIdResource' doesn't support 'IDisposable' pattern without requiring explicit cast by the caller", Scope = "type", Target = "~T:NuGet.Protocol.RemoteV3FindPackageByIdResource")]
 [assembly: SuppressMessage("Build", "CA1001:Type 'ServiceIndexResourceV3Provider' owns disposable field(s) '_semaphore' but is not disposable", Justification = "The inheritance hierarchy of 'ServiceIndexResourceV3Provider' doesn't support 'IDisposable' pattern without requiring explicit cast by the caller", Scope = "type", Target = "~T:NuGet.Protocol.ServiceIndexResourceV3Provider")]
 [assembly: SuppressMessage("Globalization", "CA1308:Normalize strings to uppercase", Justification = "We intentionally making to lower case because we're creating web api query.", Scope = "member", Target = "~M:NuGet.Protocol.PackageSearchResourceV3.SearchPage``1(System.Func{System.Uri,System.Threading.Tasks.Task{``0}},System.String,NuGet.Protocol.Core.Types.SearchFilter,System.Int32,System.Int32,NuGet.Common.ILogger,System.Threading.CancellationToken)~System.Threading.Tasks.Task{``0}")]
 [assembly: SuppressMessage("Globalization", "CA1308:Normalize strings to uppercase", Justification = "We convert packageId to lower case for api call.", Scope = "member", Target = "~M:NuGet.Protocol.PackageMetadataResourceV3.GetRegistratioIndexPageAsync(NuGet.Protocol.HttpSource,System.String,System.String,NuGet.Versioning.NuGetVersion,NuGet.Versioning.NuGetVersion,NuGet.Protocol.Core.Types.HttpSourceCacheContext,NuGet.Common.ILogger,System.Threading.CancellationToken)~System.Threading.Tasks.Task{NuGet.Protocol.Model.RegistrationPage}")]
 [assembly: SuppressMessage("Globalization", "CA1308:Normalize strings to uppercase", Justification = "We convert packageId to lower case for api call.", Scope = "member", Target = "~M:NuGet.Protocol.PackageMetadataResourceV3.LoadRegistrationIndexAsync(NuGet.Protocol.HttpSource,System.Uri,System.String,NuGet.Protocol.Core.Types.SourceCacheContext,System.Func{NuGet.Protocol.HttpSourceResult,System.Threading.Tasks.Task{NuGet.Protocol.Model.RegistrationIndex}},NuGet.Common.ILogger,System.Threading.CancellationToken)~System.Threading.Tasks.Task{System.ValueTuple{NuGet.Protocol.Model.RegistrationIndex,NuGet.Protocol.Core.Types.HttpSourceCacheContext}}")]
+[assembly: SuppressMessage("Usage", "CA2237:Mark ISerializable types with serializable", Justification = "<Pending>", Scope = "type", Target = "~T:NuGet.Protocol.PackageNotFoundProtocolException")]
```

Comment:
Why don't fix this derived Exception as well ?

Reply:
Thanks for reviewing! I tried to fix all those exceptions incrementally. This PR is for all exceptions with no properties.
Since the derived exception has it's own property and I need to add customized serialization/deserialization.

Reply:
Consider adding a message explaining why the suppression is needed, and a follow-up issue, if applicable.

```suggestion
[assembly: SuppressMessage("Usage", "CA2237:Mark ISerializable types with serializable", Justification = "Exception class with properties needs custom serialization. See https://link/to/github/issue", Scope = "type", Target = "~T:NuGet.Protocol.PackageNotFoundProtocolException")]
```